src/main.rs

```diff
@@ -207,11 +215,44 @@ fn main() -> ! {
             }
         }
 
-        let intensity = if led_pulse { pulser.intensity() } else { 1.0 };
-        led_strip.set_all(Rgb::new(
-            (led_color.0 as f32 * intensity) as u8,
-            (led_color.1 as f32 * intensity) as u8,
-            (led_color.2 as f32 * intensity) as u8,
-        ));
+        match led_pulse {
+            PulseMode::Breathing { interval_ms } => {
+                pulser.set_interval_ms(u16::from(interval_ms) as u32, &timer);
+                let intensity = pulser.intensity();
+
+                for new_led in new_leds.iter_mut() {
+                    new_led.r = intensity * led_color.0;
+                    new_led.g = intensity * led_color.1;
+                    new_led.b = intensity * led_color.2;
+                }
+            },
+            PulseMode::DialTurn => {
+                for new_led in new_leds.iter_mut() {
+                    new_led.r = 0.0;
+                    new_led.g = 0.0;
+                    new_led.b = 0.0;
+                }
+
+                new_leds[current_led].r = led_color.0;
+                new_leds[current_led].g = led_color.1;
+                new_leds[current_led].b = led_color.2;
+            },
+            PulseMode::Solid => {
+                let intensity = 1.0;
+
+                for new_led in new_leds.iter_mut() {
+                    new_led.r = intensity * led_color.0;
+                    new_led.g = intensity * led_color.1;
+                    new_led.b = intensity * led_color.2;
+                }
+            },
+        };
+
+        const FADE_CONSTANT: f32 = 0.997;
```

Comment:
I tried this out yesterday, but a `FADE_CONSTANT` of something like `0.992` or `0.993` gave more of an immediate feedback while still maintaining a nice transition between colors.